### PR TITLE
keycastr: init at 0.9.18

### PR DESCRIPTION
**Pull request by matteo-pacini**
## Description of changes

KeyCastr, an open-source keystroke visualizer.

KeyCastr enables you to share your keystrokes when creating screencasts, presenting, or collaborating with others. You can choose to display command keys, all modified keys, or all keystrokes, and there is also an option to include mouse clicks.

https://github.com/keycastr/keycastr

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 323232` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>keycastr</li>
  </ul>
</details>

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 323232` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/by-name/ke/keycastr/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,Applications}
+    cp -r KeyCastr.app $out/Applications/
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://github.com/keycastr/keycastr";
+    description = "KeyCastr, an open-source keystroke visualizer";
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/keycastr/package.nix:29]:
```suggestion
    description = "Open-source keystroke visualizer";
```

File: pkgs/by-name/ke/keycastr/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    license = lib.licenses.bsd3;
+    platforms = lib.platforms.darwin;
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/keycastr/package.nix:31]: Is this a unified app (or what do you darwin people call that)?

File: pkgs/by-name/ke/keycastr/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,Applications}
+    cp -r KeyCastr.app $out/Applications/
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/keycastr/package.nix:23]:
```suggestion
    install -Dm755 KeyCastr.app $out/Applications/
```

File: pkgs/by-name/ke/keycastr/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    hash = "sha256-q12c/W0yGoVL+wx+T/gaevx2P0Xwcv0a0FMv7bKfUnE=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,Applications}
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/keycastr/package.nix:22]:
```suggestion
```

File: pkgs/by-name/ke/keycastr/package.nix (hunk 2 above)
Comment (matteo-pacini) [reply pkgs/by-name/ke/keycastr/package.nix:31]: Darwin people got me 🤣 yes, it is an universal app, for both Intel and Apple Silicon

File: pkgs/by-name/ke/keycastr/package.nix (hunk 3 above)
Comment (matteo-pacini) [reply pkgs/by-name/ke/keycastr/package.nix:23]: Darwin apps are just folders, and `install` - as far as I'm aware - does not not perform recursive copy of folders

File: pkgs/by-name/ke/keycastr/package.nix (hunk 4 above)
Comment (matteo-pacini) [reply pkgs/by-name/ke/keycastr/package.nix:22]: See my comment: https://github.com/NixOS/nixpkgs/pull/323232#discussion_r1666045925

File: pkgs/by-name/ke/keycastr/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/ke/keycastr/package.nix:23]: Oh didn't know that, I thought it was something like appimage

**Review by Aleksanaa — APPROVED** [review]
Looks good, but I can't test it at the moment (the connection to the macbook I borrowed is lost)

File: pkgs/by-name/ke/keycastr/package.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  '';
+
+  meta = {
+    homepage = "https://github.com/keycastr/keycastr";
+    description = "Open-source keystroke visualizer";
+    license = lib.licenses.bsd3;
+    platforms = lib.platforms.darwin;
+    maintainers = with lib.maintainers; [ matteopacini ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    hydraPlatforms = [ ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ke/keycastr/package.nix:34]:
```suggestion
```
The license should reflect if we can redistribute it or not.

**Review by 4evy — APPROVED** [review]
Besides the `hydraPlatforms = [ ];` part, it LGTM!

**Comment by matteo-pacini** [comment]
@DontEatOreo @SuperSandro2000 removed `hydraPlatforms = [ ]`, thanks both for the feedback

**Pull request merged**